packages/trader/src/Modules/CFD/Components/cfd-personal-details-form.tsx

```diff
@@ -14,14 +15,152 @@ import {
     MobileWrapper,
     Modal,
     SelectNative,
-    ThemedScrollbars,
     Text,
+    ThemedScrollbars,
 } from '@deriv/components';
-import { FormSubHeader } from '@deriv/account';
 import { isDeepEqual, isDesktop, isMobile } from '@deriv/shared';
 import { Localize, localize } from '@deriv/translations';
 
-const getAccountOpeningReasonList = () => [
+type TCFDPersonalDetailsFormProps = {
+    onSave: (index: number, values: TFormValues) => void;
+    is_fully_authenticated: boolean;
+    is_loading: boolean;
+    landing_company: TLandingCompany;
+    residence_list: TResidenceObject[];
+    onCancel: () => void;
+    onSubmit: TOnSubmit;
+    value: TFormValues;
+    index: number;
+    form_error?: string;
+};
+
+type TLandingCompany = {
```

Comment:
Landing Company interface is available in `@deriv/api-types`. I think it's better to use it from there.
It's autogenerated based on our BE websocket apis.
- https://www.npmjs.com/package/@deriv/api-types

![image](https://user-images.githubusercontent.com/56330681/148891084-9d46b2bb-14d9-4d59-a2ff-6e5dd4454752.png)

Reply:
@balakrishna-binary Unfortunately, this file doesn't contain an appropriate type anymore. Currently there is no `config` or `dxtrade_financial_company` field. Perhaps, the current version suggests that all these missing fields are represented with `[k: string]: unknown;` at line 4068. It doesn't seem like a good idea to replace them with `unknown` type that can lead to facing errors such as `Property '***' does not exist on type 'unknown'` and having to anyway declare a type and apply it to the variable locally in order to be able to use it and/or its properties. I also don't intend to typify the WS response, just that exact `local` type of data that my component is expecting. What do you think? 😊

Reply:
@maryia-deriv I updated `api-types` library today. Can you pull the latest version(1.0.48) and check?
The local component is actually expecting the response from the api and there is no modification done to the response received AFAIK. So, in this case, we can utilize the type from `api-types`.